CBLClient/Apps/CBLTestServer-C/src/VectorSearchMethods.cpp

```diff
@@ -0,0 +1,280 @@
+            config.expressionLanguage = kCBLN1QLLanguage;
+            with<CBLDatabase *>(body,"database", [conn, collectionName, scopeName, indexName, config](CBLDatabase* db)
+            {
+                CBLError err;
+                CBLCollection* collection;
+                TRY(collection = CBLDatabase_CreateCollection(db, flstr(collectionName),  flstr(scopeName), &err), err);
+                TRY(CBLCollection_CreateVectorIndex(collection, flstr(indexName), config, &err), err);
+                write_serialized_body(conn, "Created index with name " + indexName);
+            });
+    }
+
+    void vectorSearch_loadDatabase(json& body, mg_connection* conn) {
+        const auto dbPath = "Databases" + string(1, DIRECTORY_SEPARATOR) + InMemoryDbName  + ".cblite2";
+        const auto dbName = InMemoryDbName;
+        char cwd[1024];
+        cbl_getcwd(cwd, 1024);
+        const auto databasePath = string(cwd) + DIRECTORY_SEPARATOR + dbPath;
+        const auto extensionsPath = string(cwd) + DIRECTORY_SEPARATOR + APP_EXTENSIONS_DIR;
+        CBL_SetExtensionPath(flstr(extensionsPath));
+        CBLDatabaseConfiguration* databaseConfig = nullptr;
+        CBLError err;
+        CBLDatabase* db;
+        TRY(CBL_CopyDatabase(flstr(databasePath), flstr(dbName), databaseConfig, &err), err);
+        wordMap = getWordMap();
+        TRY(db = CBLDatabase_Open(flstr(dbName), databaseConfig, &err), err);
+        write_serialized_body(conn, memory_map::store(db, CBLDatabase_EntryDelete));
+
+    }
+   
+   void vectorSearch_registerModel(json& body, mg_connection* conn) {
+        const auto name = body["name"].get<string>();
+        const auto key = body["key"].get<string>();
+
+        CBLPredictiveModel model = {};
+        model.context = nullptr;
+        model.prediction = predictFunction;
+        CBL_RegisterPredictiveModel(flstr(name), model);
+        write_serialized_body(conn, "Model registered");
+    }
+    
+    void vectorSearch_getEmbedding(json& body, mg_connection* conn)
+    {
+        auto embbedingsVector =  FLMutableArray_New();
+        DEFER {
+            FLMutableArray_Release(embbedingsVector);
+        };
+        auto vectorDict = getEmbeddingDic(body["input"].get<string>());
+        FLValue embeddings = FLDict_Get(vectorDict, flstr("vector"));
+        FLArrayIterator iter;
+        FLArrayIterator_Begin(FLValue_AsArray(embeddings), &iter);
+        FLValue value;
+        while (NULL != (value = FLArrayIterator_GetValue(&iter))) {
+            FLMutableArray_AppendFloat(embbedingsVector, FLValue_AsFloat(value));
+            FLArrayIterator_Next(&iter);
+        }
+        write_serialized_body(conn, (FLValue)embbedingsVector);
+    }
+
+
+    void vectorSearch_query(json& body, mg_connection* conn) {
+
+        with<CBLDatabase *>(body,"database", [conn, body](CBLDatabase* db)
+            {
+                auto embeddedTermDic = getEmbeddingDic(body["term"].get<string>());
```

Comment:
Need to release embeddedTermDic as well.

Reply:
Thanks, fixed